pkg/monitor/monitor/util.go

```diff
@@ -272,31 +316,76 @@ chmod 777 /data/prometheus /data/grafana
 		})
 	}
 
-	if monitor.Spec.Grafana != nil {
-		container.VolumeMounts = append(container.VolumeMounts, core.VolumeMount{
-
-			MountPath: "/etc/grafana/provisioning/datasources",
-			Name:      "datasource",
-			ReadOnly:  false,
-		}, core.VolumeMount{
-
-			MountPath: "/grafana-dashboard-definitions/tidb",
-			Name:      "grafana-dashboard",
-			ReadOnly:  false,
+	getGrafanaEnvs(monitor, &container)
+	var envOverrides []core.EnvVar
+	for k, v := range monitor.Spec.Initializer.Envs {
+		envOverrides = append(envOverrides, core.EnvVar{
+			Name:  k,
+			Value: v,
 		})
-		container.Env = append(container.Env,
-			core.EnvVar{
-				Name:  "GF_PROVISIONING_PATH",
-				Value: "/grafana-dashboard-definitions/tidb",
+	}
+	container.Env = util.AppendOverwriteEnv(container.Env, envOverrides)
+	return container
+}
+
+func getMonitorDMInitContainer(monitor *v1alpha1.TidbMonitor, dc *v1alpha1.DMCluster) core.Container {
+	command := getInitCommand(monitor)
+	dmImageVersion := dc.MasterImage()
+	container := core.Container{
+		Name:  "monitor-dm-initializer",
+		Image: fmt.Sprintf("%s:%s", monitor.DMSpec.Initializer.BaseImage, monitor.DMSpec.Initializer.Version),
+		Env: []core.EnvVar{
+			{
+				Name:  "DM_CLUSTER_NAME",
+				Value: dc.Name,
 			},
-			core.EnvVar{
-				Name:  "GF_DATASOURCE_PATH",
-				Value: "/etc/grafana/provisioning/datasources",
-			})
+			{
+				Name:  "PROM_CONFIG_PATH",
+				Value: "/prometheus-rules",
+			},
+			{
+				Name:  "PROM_PERSISTENT_DIR",
+				Value: "/data",
+			},
+			{
+				Name:  "DM_VERSION",
```

Comment:
How is this used?

Reply:
I have change the value of this envVar to the image version of tidb, but i reserve the name of this envVar: "DM_VERSION", so we do not need th change dm init [image](https://github.com/pingcap/dm/blob/master/monitoring/init.sh#L24). We could support reloading rules of dm in reloader in the future.

Reply:
Then we can just keep the value you set here, no need to change it to the tidb image version.
BTW, did you test the case to modify the alert rules of dm-worker and see if they are persisted in the dir named with tidb image version?

Reply:
I have tested this case. After changing dm rules in reloader, the `dm_worker.rules.yml` is persisted in `/data/latest-rules/tidb:v4.0.8/`.